packages/dialog/src/react/index.tsx

```diff
@@ -0,0 +1,169 @@
+import { compose, css, keyframes } from 'glamor'
+import React, { HTMLAttributes } from 'react'
+
+import FocusManager from '@pluralsight/ps-design-system-focusmanager'
+import Theme from '@pluralsight/ps-design-system-theme'
+import {
+  ValueOf,
+  RefForwardingComponent,
+  isFunction
+} from '@pluralsight/ps-design-system-util'
+
+import stylesheet from '../css'
+import * as vars from '../vars'
+
+const MODAL_OVERLAY_ID = 'psds-dialog__overlay'
+
+const fade = keyframes(stylesheet['@keyframes psds-dialog__keyframes__fade'])
+
+const styles = {
+  dialog: (modal: boolean, tailPosition?: ValueOf<typeof vars.tailPositions>) =>
+    compose(
+      css(stylesheet['.psds-dialog']({ fade })),
+      modal && css(stylesheet['.psds-dialog--modal']),
+      Boolean(tailPosition) &&
+        compose(
+          stylesheet['.psds-dialog--w-tail'],
+          stylesheet[`.psds-dialog--tailPosition-${tailPosition}`]
+        )
+    ),
+  content: (style: Record<string, unknown> = {}) =>
+    css(stylesheet['.psds-dialog__content'], style),
+  close: () => css(stylesheet['.psds-dialog__close']),
+  overlay: () => css(stylesheet['.psds-dialog__overlay'])
+}
+
+const CloseButton: React.FC<HTMLAttributes<HTMLButtonElement>> = props => (
+  <button {...styles.close()} {...props} aria-label="Close dialog">
+    <svg
+      aria-label="close icon"
+      role="img"
+      viewBox="0 0 24 24"
+      xmlns="http://www.w3.org/2000/svg"
+    >
+      <path d="M18 7.41L16.59 6 12 10.59 7.41 6 6 7.41 10.59 12 6 16.59 7.41 18 12 13.41 16.59 18 18 16.59 13.41 12" />
+    </svg>
+  </button>
+)
+
+interface OverlayProps extends HTMLAttributes<HTMLDivElement> {
+  'aria-label'?: string
+  disableCloseOnOverlayClick?: boolean
+  onClose?: (evt: React.MouseEvent) => void
+}
+
+const Overlay: React.FC<OverlayProps> = props => {
+  function handleOverlayClick(evt: React.MouseEvent) {
+    if (props.disableCloseOnOverlayClick) return
+    if ((evt.target as HTMLDivElement).id === MODAL_OVERLAY_ID)
+      props.onClose && props.onClose(evt)
+  }
+
+  return (
+    <div
+      {...styles.overlay()}
+      {...props}
+      id={MODAL_OVERLAY_ID}
+      onClick={handleOverlayClick}
+      role="region"
+    />
+  )
+}
+
+export interface DialogStatics {
+  tailPositions: typeof vars.tailPositions
+}
+export interface DialogProps extends HTMLAttributes<HTMLDivElement> {
+  'aria-label'?: string
+  disableCloseButton?: boolean
+  disableCloseOnEscape?: boolean
+  disableCloseOnOverlayClick?: boolean
+  disableFocusOnMount?: boolean
+  modal?: boolean
+  onClose?: (evt: React.KeyboardEvent | React.MouseEvent) => void
+  tailPosition?: ValueOf<typeof vars.tailPositions>
+  returnFocus?: boolean
+  // eslint-disable-next-line camelcase
+  UNSAFE_stylesFor?: Record<string, unknown>
```

Comment:
I think this is ok for now.  

I'm imagining how difficult this may be for devs to use.  Each one of these styles objects is totally custom (though we have a small convention).  So, it could be nice to have some typings at at least those first level object keys so that developers can have some help constructing the theme objects.

Reply:
I don't think a typing would matter here...it's an eslint error not a TS one. Typescript doesn't care. Does typescript override eslint if we make a type?

Reply:
I agree with what you said: A typing, as I suggested, wouldn't affect eslint.  I wasn't thinking about eslint.  

Also to clarify: this isn't something I want necessarily included in the "request for change" on the PR.  Was me more pondering the future.

I was just thinking about how a developer might benefit from a type more specific than `Record<string, unknown>`.